pc/registration/icp/evaluator.go

```diff
@@ -102,22 +122,21 @@ func (e *PointToPointEvaluator) Evaluate(base storage.Search, target pc.Vec3Rand
 		pairVecs[0].(pc.Vec3Slice)[i] = pb
 		pairVecs[1].(pc.Vec3Slice)[i] = pt
 
+		w := weightFn(pair.SquaredDistance)
```

Comment:
[optional] Do you think it could be useful to just pass the pair as argument to the weight function?I don't have any clear example but we could imagine some weight function using something else than the squared distance to assign a weight to a point pair.

Reply:
This was just a thought so no need to make any change in response but the squared distance could a property of the pair if we want to avoid recomputing it in the weight function while passing the pair as an argument.